MyApplication/app/src/main/java/com/courses/bogdan/myapplication/activities/MainActivity.java

```diff
@@ -15,86 +17,164 @@
+    }
 
-        button.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                String s = String.valueOf(textView.getText());
-                Log.e("TAAAG", s);
-                Toast.makeText(MainActivity.this,"i pressed the buttoon", Toast.LENGTH_SHORT).show();
-            }
-        });
+    @Override
+    protected void onStart() {
+        super.onStart();
+        presenter.onViewStarted();
+    }
 
-        getButton.setOnClickListener(new View.OnClickListener(){
-            @Override
-            public void onClick(View view) {
-                textView.setText(editText.getText());
-            }
-        });
+    @Override
+    protected void onResume() {
+        super.onResume();
+        registerBroadcastReceiver();
+    }
 
-        colorButton.setOnClickListener(new View.OnClickListener(){
-            @Override
-            public void onClick(View view) {
-                Random randomColor = new Random();
-                red= randomColor.nextInt(256);
-                green = randomColor.nextInt(256);
-                blue = randomColor.nextInt(256);
-                layout.setBackgroundColor(Color.argb(255,red,green,blue));
+    @Override
+    protected void onPause() {
+        super.onPause();
+        unregisterReceiver(myReceiver);
+    }
 
+    private void registerBroadcastReceiver() {
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(IntentActionConstants.ACTIVITY_INFO_INTENT);
+        registerReceiver(myReceiver, intentFilter);
+    }
+
+    private void setupViews() {
+        mainlayout = (RelativeLayout) findViewById(R.id.layout);
+        toastButton = (Button) findViewById(R.id.toastButton);
+        getButton = (Button) findViewById(R.id.getTextButton);
+        colorButton = (Button) findViewById(R.id.colorButton);
+        fragmentButton = (Button) findViewById(R.id.fragmentGo);
+        myView = (TextView) findViewById(R.id.myText);
+        textEditor = (EditText) findViewById(R.id.textEditor);
+
+        toastButton.setOnClickListener(toastButtonListener);
+        getButton.setOnClickListener(getButtonListener);
+        colorButton.setOnClickListener(colorButtonListener);
+        fragmentButton.setOnClickListener(fragmentButtonListener);
+    }
 
+    private void setupFragment() {
+        MainFragment fragment = MainFragment.newInstance();
+        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+        transaction.add(R.id.main_frame_layout, fragment).commit();
+    }
+
+    private void setupListeners() {
+        colorButtonListener = new ColorButtonListener();
+        fragmentButtonListener = new FragmentButtonListener();
+        getButtonListener = new GetButtonListener();
+        toastButtonListener = new ToastButtonListener();
+    }
+
+    private void setTextToTextView(final String text) {
+        this.runOnUiThread(new Runnable() {
```

Comment:
Aici poti lasa doar linia aceasta: myView.setText(text); in metoda deoarece aceasta operatiune oricum se efectueaza pe threadul de UI. :)